helm/cluster-vsphere/templates/copy-secret-hook-job.yaml

```diff
@@ -0,0 +1,134 @@
+    {{- include "labels.common" $ | nindent 4 }}
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["create"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: "{{ include "resource.default.name" $ }}-read-secret"
+  namespace: "{{ .Values.userContext.secretRef.secretNamespace }}"
+  annotations:
+    {{- include "copyHookAnnotations" $ | nindent 4 }}
+    "helm.sh/hook-weight": "-2"
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+subjects:
+  - kind: ServiceAccount
+    name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+    namespace: "{{ $.Release.Namespace }}"
+roleRef:
+  kind: Role
+  name: "{{ include "resource.default.name" $ }}-read-secret"
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: "{{ include "resource.default.name" $ }}-create-secret"
+  namespace: "{{ .Values.userContext.secretRef.secretNamespace }}"
+  annotations:
+    {{- include "copyHookAnnotations" $ | nindent 4 }}
+    "helm.sh/hook-weight": "-2"
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+subjects:
+  - kind: ServiceAccount
+    name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+    namespace: "{{ $.Release.Namespace }}"
+roleRef:
+  kind: Role
+  name: "{{ include "resource.default.name" $ }}-create-secret"
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+  namespace: "{{ $.Release.Namespace }}"
+  annotations:
+    {{- include "copyHookAnnotations" $ | nindent 4 }}
+    "helm.sh/hook-weight": "-1"
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+spec:
+  ttlSecondsAfterFinished: 86400 # 24h
+  template:
+    metadata:
+      name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+      namespace: "{{ $.Release.Namespace }}"
+      labels:
+        {{- include "labels.common" $ | nindent 8 }}
+    spec:
+      restartPolicy: Never
+      serviceAccountName: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+      securityContext:
+        runAsUser: {{ include "securityContext.runAsUser" $ }}
+        runAsGroup: {{ include "securityContext.runAsGroup" $ }}
+      containers:
+        - name: copy-secret-hook
+          image: "{{ .Values.kubectlImage.registry }}/{{ .Values.kubectlImage.name }}:{{ .Values.kubectlImage.tag }}"
+          command:
+            - "/bin/sh"
+            - "-xc"
+            - |
+              # _cred=$(kubectl get secret {{ .Values.userContext.secretRef.secretName }} -n "{{ .Values.userContext.secretRef.secretNamespace }}" -ojson |\
+              #          jq -r '.data | "\(.username | @base64d)\n\(.password | @base64d)"')
+              # [ $? -ne 0 ] && echo "can't get the credentials" >&2 && exit 1
+              # _username=$(echo $_cred | head -1)
+              # _password=$(echo $_cred | tail -1)
```

Comment:
these 5 lines above can either go away or can be used if we decide that we want these values in plain text to pass them somehow to CPI&CSI chart

Hard part here is that the secrets for CPI&CNI need to be created in a different cluster. This could be done in a post-install hook (not to block the cluster creation), so that similar secret would be created in WC from which the CNI & CPI helm charts can read/get what they need. These hooks adds more complexity though, on the other hand we will not have the credentials as part of the value files.